src/Pagination.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+
+export default class Pagination extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = this._calculatePages(props);
+
+    // This binding is necessary to make `this` work in the callback
+    // https://facebook.github.io/react/docs/handling-events.html
+    this._handleSubmit = this._handleSubmit.bind(this);
+    this._updateInputValue = this._updateInputValue.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState(this._calculatePages(nextProps));
+  }
+
+  _calculatePages(props) {
+    const from = props.from || 0;
+    const size = props.size || 1;
+    const total = props.total || 0;
+    const c = Math.ceil(from / size) + 1;
+
+    return {
+      current: c,
+      total: Math.ceil(total / size),
+      inputValue: c
+    }
+  }
+
+  _handleSubmit(event) {
+    event.preventDefault();
+    this._setPage(this.state.inputValue);
+  }
+
+  _setPage(page) {
+    this.props.onPage(page);
+  }
+
+  _updateInputValue(evt) {
+    this.setState({
+      inputValue: evt.target.value
+    });
+  }
+
+  render() {
+    if( this.props.total === 0) {
+      return null;
+    }
+
+    return (
+      <nav className="m-pagination"
+           role="navigation"
+           aria-label="Pagination">
+          <button className="a-btn
+                    a-btn__icon-on-left
+                    cf-icon
+                    cf-icon-left
+                    cf-icon__before
+                    m-pagination_btn-prev"
+                    onClick={() => this._setPage(this.state.current - 1)}
+                    disabled={this.state.current <= 1}>
+              Previous
+          </button>
+          <button className="a-btn
+                    a-btn__icon-on-right
+                    cf-icon
+                    cf-icon-right
+                    cf-icon__after
+                    m-pagination_btn-next"
+                    onClick={() => this._setPage(this.state.current + 1)}
+                    disabled={this.state.current >= this.state.total}>
+              Next
+          </button>
```

Comment:
I believe that if you update these buttons to the [latest button-with-icon markup](https://cfpb.github.io/capital-framework/components/cf-buttons/#icon-buttons), you'll get rid of that unsightly lack of space and divider between the icon and the text.